tgui/packages/tgui/interfaces/SDQLSpellMenu.js

```diff
@@ -0,0 +1,476 @@
+                      content="Confirm"
+                      confirmContent="Are you sure?"
+                      onClick={() => act('confirm')} />
+                    <Button
+                      disabled={!type}
+                      tooltip="Save the spell to a json file on your local system."
+                      onClick={() => act('save')}>
+                      Save Spell
+                    </Button>
+                    <Button
+                      tooltip="Load a spell from a json file on your local system."
+                      onClick={() => act('load')}>
+                      Load Spell
+                    </Button>
+                  </Stack.Item>
+                  <Stack.Item grow basis={0} />
+                  <Stack.Item textColor="bad">
+                    {alert}
+                  </Stack.Item>
+                </Stack>
+              </Stack.Item>
+            </Stack>
+          </Stack.Item>
+          <Stack.Item minWidth="128px">
+            <SDQLSpellIcons />
+          </Stack.Item>
+        </Stack>
+      </Window.Content>    
+    </Window>
+  );
+};
+
+const var_condition = (entry, context) => {
+  const { data } = useBackend(context);
+  const {
+    saved_vars,
+  } = data;
+  switch (entry.name) {
+    case "charge_max":
+      return (Object.prototype.hasOwnProperty.call(saved_vars, "charge_type") && saved_vars["charge_type"]) !== "holder_var";
+    case "holder_var_type":
+    case "holder_var_amount":
+      return (Object.prototype.hasOwnProperty.call(saved_vars, "charge_type") && saved_vars["charge_type"]) === "holder_var";
+    case "human_req":
+      return (Object.prototype.hasOwnProperty.call(saved_vars, "clothes_req") && !saved_vars["clothes_req"]);
+    case "invocation":
+      return (Object.prototype.hasOwnProperty.call(saved_vars, "invocation_type") && saved_vars["invocation_type"]) !== "none";
+    case "invocation_emote_self":
+      return (Object.prototype.hasOwnProperty.call(saved_vars, "invocation_type") && saved_vars["invocation_type"]) === "emote";
+    case "overlay_icon":
+    case "overlay_icon_state":
+    case "overlay_lifespan":
+      return (Object.prototype.hasOwnProperty.call(saved_vars, "overlay") && saved_vars["overlay"]);
+    case "sparks_amt":
+      return (Object.prototype.hasOwnProperty.call(saved_vars, "sparks_spread") && saved_vars["sparks_spread"]);
+    case "smoke_amt":
+      return (Object.prototype.hasOwnProperty.call(saved_vars, "smoke_spread") && saved_vars["smoke_spread"]);
+    case "random_target_priority":
+      return (Object.prototype.hasOwnProperty.call(saved_vars, "random_target") && saved_vars["random_target"]);
+    default:
+      return true;
+  }
+};
+
+const WrapInTooltip = (props, context) => {
+  const { data } = useBackend(context);
+  const { entry } = props;
+  const {
+    type,
+    tooltips,
+  } = data;
+  const tip = tooltips[entry.name]?.replace("$type", tooltips[(entry.name+"_"+type) || "Something went wrong."]);
```

Comment:
"Something went wrong." is dead code.

```suggestion
  const tip = tooltips[entry.name]?.replace("$type", tooltips[entry.name + "_" + type]);
```